tests/helpers/test_event.py

```diff
@@ -163,6 +164,91 @@ async def wildercard_run_callback(entity_id, old_state, new_state):
     assert len(wildercard_runs) == 6
 
 
+async def test_async_track_state_change_event(hass):
+    """Test async_track_state_change_event."""
+    single_entity_id_tracker = []
+    multiple_entity_id_tracker = []
+
+    @ha.callback
+    def single_run_callback(event):
+        old_state = event.data.get("old_state")
+        new_state = event.data.get("new_state")
+
+        single_entity_id_tracker.append((old_state, new_state))
+
+    @ha.callback
+    def multiple_run_callback(event):
+        old_state = event.data.get("old_state")
+        new_state = event.data.get("new_state")
+
+        multiple_entity_id_tracker.append((old_state, new_state))
+
+    unsub_single = async_track_state_change_event(
+        hass, ["light.Bowl"], single_run_callback
+    )
+    unsub_multi = async_track_state_change_event(
+        hass, ["light.Bowl", "switch.kitchen"], multiple_run_callback
+    )
+
+    # Adding state to state machine
+    hass.states.async_set("light.Bowl", "on")
+    await hass.async_block_till_done()
+    assert len(single_entity_id_tracker) == 1
+    assert single_entity_id_tracker[-1][0] is None
+    assert single_entity_id_tracker[-1][1] is not None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)